runtime/src/main/kotlin/kotlin/text/Strings.kt

```diff
@@ -206,7 +206,7 @@ public actual fun CharSequence.repeat(n: Int): String {
         else -> {
             when (length) {
                 0 -> ""
-                1 -> this[0].let { char -> fromCharArray(CharArray(n) { char }, 0, n) }
+                1 -> this[0].let { char -> unsafeFromCharArray(CharArray(n) { char }, 0, n) }
```

Comment:
We can use the public API `CharArray.concatToString()` (the one for entire array) in such cases. It does the same under the hood.

Reply:
+1